packages/data-broker/src/DataBroker.js

```diff
@@ -241,14 +249,70 @@ export class DataBroker {
 
   mergeSyncGroups = (target = [], source) => target.concat(source);
 
-  async pullMetadata(dataSourceSpec, options) {
+  async pullMetadata(dataSourceSpec, options = {}) {
     const dataSources = await this.fetchDataSources(dataSourceSpec);
-    if (countDistinct(dataSources, 'service_type') > 1) {
+    const { type: dataSourceType } = dataSourceSpec;
+    const { organisationUnitCode } = options;
+
+    const dataSourcesByServiceType = await this.resolveServiceType(
+      dataSources,
+      organisationUnitCode,
+    );
+    if (Object.keys(dataSourcesByServiceType).length > 1) {
       throw new Error('Cannot pull metadata for data sources belonging to different services');
     }
-    const service = this.createService(dataSources[0].service_type);
-    // `dataSourceSpec` is defined  for a single `type`
-    const { type } = dataSourceSpec;
-    return service.pullMetadata(dataSources, type, options);
+
+    const nestedResults = await Promise.all(
+      Object.entries(dataSourcesByServiceType).map(([serviceType, dataSources]) =>
+        this.createService(serviceType).pullMetadata(dataSources, dataSourceType, options),
+      ),
+    );
+
+    return nestedResults.reduce(
+      (results, resultsForService) => results.concat(resultsForService),
+      [],
+    );
+  }
+
+  /**
+   * Resolves the service type of each data source and groups by service type
+   * @param {DataElement[]|DataGroup[]} dataSources
+   * @param {string} [orgUnitCode]
+   * @returns {Object<string, DataElement[]|DataGroup[]>} map of serviceType -> data sources
+   * @private
+   */
+  async resolveServiceType(dataSources, orgUnitCode) {
+    const map = {};
+    const addToMap = (k, v) => {
+      if (!map[k]) map[k] = [];
+      map[k].push(v);
+    };
+
+    const entity = orgUnitCode ? await this.models.entity.findOne({ code: orgUnitCode }) : null;
+    const { countryCode } = entity ?? {};
```

Comment:
Do we want to throw an error here if we can't find an entity/countryCode? Feels a tad unsafe that we could follow through to to the below logic with a undefined countryCode.

Bit on the fence but just wanted to raise it